src/env.rs

```diff
@@ -260,10 +268,16 @@ impl Source for Environment {
                     }
                 } else {
                     // Skip this key
-                    return;
+                    return Ok(());
                 }
             }
 
+            // At this point, we don't know if the key is required or not.
+            // Therefore if the value is not a valid unicode string, we error out.
+            let value = value.into_string().map_err(|os_string| {
+                ConfigError::Message(format!("key {key:?} is not valid Unicode: {os_string:?}"))
```

Comment:
Since we already are making a change, we should fix this error message.  THis implies the key is invalid and not the value it points to.  We also don't give the context for this being an env variable.

maybe something like "env variable `{key}` contains non-UTF 8 data: {osstring:?}"

Reply:
Np. Shouldn't it remain `Unicode` and not `UTF-8`? I deliberately changed all references to `Unicode` because [OsStr.to_str](https://doc.rust-lang.org/std/ffi/struct.OsStr.html#method.to_str) uses this term.

Reply:
Sure

Reply:
I also went with `{key:?}` so it's properly quoted.